core/lib/instrument_api.h

```diff
@@ -4553,6 +4553,22 @@ DR_API
 bool
 dr_raw_tls_cfree(uint offset, uint num_slots);
 
+/**
+ * Provides raw access to offset \p offset from TLS base \p tls_register
+ * and writes value \p value.
+ */
+DR_API
+void
+dr_raw_write_tls_slot(reg_id_t *tls_register, ushort offset, void *value);
```

Comment:
The existing routines analogous to this (dr_get_tls_field, drmgr_get_tls_field) take in the dcontext and target that thread instead of the current thread.  Some use cases require that.  That may be tricky to do here.  If so the docs should make it clear this is only the current thread (yes I'd do that even w/o taking in dcontext).

Reply:
Done, added to description.